### ISSUE TITLE
Split room list by subspace

### ISSUE THREAD

**Issue by frandavid100** [element-hq/element-web#17378]
This is what a regular room list looks like in Discord:

![image](https://user-images.githubusercontent.com/694794/118780528-70109800-b88c-11eb-9b1d-c74cca4a3b8d.png)

After trying to bridge some Discord communities, their members complain that the room list looks disorganized in Element because the rooms are not thematically organized (like in the screenshot; where there are Info, Support, Conversation and Gaming categories) and instead there is a single list, containing unrelated rooms.

Instead, the room list could be grouped by subspace like this:

![image](https://user-images.githubusercontent.com/694794/118782739-b2d36f80-b88e-11eb-957f-0b4b00c7b922.png)

I think this would make it more manageable, and that it would make rooms easier to find.

**Comment by t3chguy** [comment]
It makes it much harder to track unread rooms, I often struggle with Discord because of this (especially given you can't really collapse sublists)

Imagine if you had ~10 rooms in each of those lists you proposed. `Conversation` would be scrolled off the bottom and you wouldn't see any notification badges in it

**Comment by frandavid100** [comment]
> It makes it much harder to track unread rooms

Thanks for the reply. I took that into account and added a "Notifications" section on top, where all the unread rooms in that space (including those in subspaces) would appear, in order to make them noticeable and easy to reach.

Would that be acceptable?

**Comment by t3chguy** [comment]
Personally I wouldn't like it, but I am just one person. I prefer a single list sorted by my preference rather than lots of little ones which end up messier for me

**Comment by SimonBrandner** [comment]
I kind of have an argument with myself. We should make a decision between hard-ordering and auto-ordering (by notifications/A-Z) and I think the combination of the two might get confusing

**Comment by frandavid100** [comment]
> Personally I wouldn't like it, but I am just one person. I prefer a single list sorted by my preference rather than lots of little ones which end up messier for me

I mean if that would solve the issue you mentioned, where unread rooms would be hard to find.

**Comment by t3chguy** [comment]
@frandavid100 not quite for me given depending on what counts as "unread" it'd either be too noisy or not noisy enough. I have a lot of my rooms on Mentions & Keywords, some of which I care when there is an unread but not notification, others not

**Comment by frandavid100** [comment]
> This would mean you could have the same room in the room-list twice, which just is... not nice

I assume you mean that, if a room is on two subspaces, it would be shown twice on the list. That wouldn't necessarily be the case: the list could only show the room once, in the subspace at the top, and then not show it again even if it's in other subspaces.

This is all assuming subspaces can be reordered manually, and more important ones can be dragged to the top.

**Comment by t3chguy** [comment]
subspace ordering is defined by the admin only

**Comment by frandavid100** [comment]
> @frandavid100 not quite for me given depending on what counts as "unread" it'd either be too noisy or not noisy enough. I have a lot of my rooms on Mentions & Keywords, some of which I care when there is an unread but not notification, others not

I understand. I guess it has both upsides and downsides.

That said; take into account that Matrix would give you several ways to deal with that problem, that you don't have on Discord. For example you could just not join any rooms in subspaces that don't interest you (which would mean that whole section wouldn't even appear on the list); or if you're very interested in specific subspaces you could join only those instead of the main space.

Not saying those options don't have problems of their own. Just saying they exist on Matrix, and not on Discord.

**Comment by t3chguy** [comment]
Right, except in my case I'm interesting in receiving mentions in them, but not otherwise

**Comment by SimonBrandner** [comment]
> I assume you mean that, if a room is on two subspaces, it would be shown twice on the list. That wouldn't necessarily be the case: the list could only show the room once, in the subspace at the top, and then not show it again even if it's in other subspaces.

Well, I find both cases unideal...

**Comment by Appenzeill** [comment]
I feel that this would help compartmentalize a server into sections. 
Personally I like this feature from Discord, but I can see why people would dislike it.
I use my Discord Communities as a place to hangout with friends, so being able to make distinct sections for different tasks is really useful to me. 
I have a section for memes and general talk, but also multiple sections for different development projects. Being able to give those a separate compartment is really nice for that. It prevents different projects from cluttering each others compartments.

**Comment by alyaeanyx** [comment]
Perhaps, a suitable solution to this problem would be to keep the current model of spaces/subspaces and lists ordered according to the user's preference, but to filter out rooms belonging to a subspace when viewing the higher-level space. Currently, the spaces view recursively searches for and shows all rooms in all subspaces, and I think that it would be a lot more convenient if the view of a single space would only include the direct member rooms of that space, similar to filesystem hierachies. Then, you could also easily navigate to an individual notification with the subspaces menu.

**Comment by frandavid100** [comment]
> Then, you could also easily navigate to an individual notification with the subspaces menu

Ideally, I would like to avoid interacting with the subspaces menu as much as possible. So I'm opposed to this.

**Comment by SimonBrandner** [comment]
> > Then, you could also easily navigate to an individual notification with the subspaces menu
> 
> Ideally, I would like to avoid interacting with the subspaces menu as much as possible. So I'm opposed to this.

I partially agree. I sometimes want to interact with subspaces but sometimes I don't. Not showing all sub-rooms would make this hard for me

**Comment by frandavid100** [comment]
This whole thing is really hard to agree on 😑

**Comment by BohdanOpyr** [comment]
Why not add one more sorting option? "A-Z", "Activity" and "Categorical" or whatever?

**Comment by SimonBrandner** [comment]
> Why not add one more sorting option? "A-Z", "Activity" and "Categorical" or whatever?

Because the sorting `A-Z` and `Activity` is a setting for a specifc sublist not the whole room-list

**Comment by HansJK** [comment]
This is a must in order to keep large spaces well organised!
You could also add tiny subspace icon to easier tell it apart:
![Show Rooms from Subspaces](https://user-images.githubusercontent.com/37002225/119722503-ad68bd00-be6c-11eb-9298-934ab75c9678.png)
This is so much cleaner than an endless list of random rooms like this:
![Screenshot 2021-05-26 at 21 37 22](https://user-images.githubusercontent.com/37002225/119722515-b063ad80-be6c-11eb-8a8c-6a4a619ecee2.png)

**Comment by luxaritas** [comment]
To add on to the points about being able to see unread messages, Discord and Slack actually have potential paradigms for this already. One method would be to have a badge of some sort on the bottom of the list indicating "scroll down, there's more activity below." The other would be to move unread conversations to a separate section on the top.

FWIW, I'll add a potential use case. We've established that a large room list is difficult to manage. That can be somewhat alleviated if you categorize your rooms into multiple spaces, and then you drill down by space. However, if you have such nested spaces with lots of rooms, if you create a room on a top-level room, we're back to the original problem of potentially not being able to find the room amidst the giant list. This means you _have_ to create rooms as "leaves" in spaces.

**Comment by anton-molyboha** [comment]
There is a similar discussion in #18905 ,

**Comment by Jo-Blade** [comment]
> To add on to the points about being able to see unread messages, Discord and Slack actually have potential paradigms for this already. One method would be to have a badge of some sort on the bottom of the list indicating "scroll down, there's more activity below."

Yes I totally need this personally because I use matrix for my association with a lot of subspaces like «bookkeeping», «new projects»… and it is very difficult today to read all rooms because there are too many, and they are all listed in the main space without categorizing.
I very often have a button «show 31 rooms more» that is very unpractical. I really prefer the hierarchy system of cinny client for this time, and personally I am a fan of the discord approach. I think it could be a good thing if element add something like @HansJK proposed.

**Comment by t3chguy** [comment]
@rufuskahler I believe your designs address this
